components/usage/pkg/db/billed_session.go

```diff
@@ -27,3 +29,23 @@ type BilledSession struct {
 func (d *BilledSession) TableName() string {
 	return "d_b_billed_session"
 }
+
+func SetBilled(ctx context.Context, conn *gorm.DB, instanceID uuid.UUID, instanceCreationTime time.Time, system string) error {
+	database := conn.WithContext(ctx)
+
+	billedSession := BilledSession{InstanceID: instanceID, From: NewVarcharTime(instanceCreationTime), System: system}
+	db := database.Create(&billedSession)
+
+	return db.Error
+}
+
+func GetBilled(ctx context.Context, conn *gorm.DB, instanceID uuid.UUID) ([]BilledSession, error) {
+	var billedSessions []BilledSession
+	db := conn.WithContext(ctx).Where("InstanceID = ?", instanceID).Find(&billedSessions)
+
+	if db.Error != nil {
+		return nil, fmt.Errorf("Not found for instance ID: %w", db.Error)
```

Comment:
```suggestion
		return nil, fmt.Errorf("not found for instance ID: %w", db.Error)
```
error strings shouldn't be capitalised in Go. Also could be improved to say *what* wasn't found.